### DISCUSSION TITLE
Preview builds site fine but render and publish fail

### DISCUSSION THREAD

**Discussion by AustenLamacraft** [quarto-dev/quarto-cli#4033] [Q&A]
I'm finding that preview works great: I have HTML in `_book` and it looks great. But then `render` and `publish` (which presumably calls `render`) both try and do something with xelatex and then fail, producing an incomplete pdf in the process.

I'm not sure what's going on. Why does `render` do this when `preview` makes HTML perfectly OK? Any help would be great!

**Comment by cderv** [comment]
To what format are you rendering ? What is in your `_quarto.yml` ? 

`quarto render` will by default render to all format. Publish will do too. See here: https://quarto.org/docs/books/#publishing

Are you able to render you book project to PDF ? 
`quarto render --to pdf` ? This could be failling for several reasons depending for example on the LaTeX environment you have installed, and LaTeX packages available. 

Without more information, it will be hard to help further.

**Comment by AustenLamacraft** [comment]
Ah OK, thanks. I want to render to html only (for now) so I guess I should do `quarto render --to html`? But I was hoping to use `quarto publish` to publish directly to Netlify without Quarto trying to make a PDF. Rendering then pushing to GH and then serving Netlify out of `_book` is OK though

**Reply by cscheid** [reply]
You can use project profiles in the `quarto publish` command, and [project profiles](https://quarto.org/docs/projects/profiles.html) allow you to choose different formats.

```
$ quarto publish --help

  Usage:   quarto publish [provider] [path]
...

  Options:

    -h, --help              - Show this help.
...
    --profile               - Active project profile(s)

  Commands:

    help  [command]  - Show this help or the help of a sub-command.

  Examples:

    Publish project (prompt for provider):  quarto publish
...
```

**Reply by AustenLamacraft** [reply]
Thanks! I think I've got it now. I had `format: pdf` specified in my `_quarto.yml` so I understand now that it builds a pdf in that case even if I'm just using `quarto publish`.
